octree/include/pcl/octree/octree2buf_base.h

```diff
@@ -94,7 +94,18 @@ class BufferedBranchNode : public OctreeNode {
   inline OctreeNode*
   getChildPtr(unsigned char buffer_arg, unsigned char index_arg) const
   {
-    assert((buffer_arg < 2) && (index_arg < 8));
+    if (buffer_arg >= 2) {
```

Comment:
I would prefer `assert` over `if` in all three functions here (`getChildPtr`, `setChildPtr`, `hasChild`). You can add hints to `buffer_arg` and `index_arg` documentation if you like